### PR TITLE
Switch to clover

### PR DESCRIPTION
**Pull request by dogeared**
* Switches testing library to clover
* Update build (`.travis.yml`) to run tests with clover
* Temporarily host coveralls maven plugin snapshot version with clover support
* Generate coveralls data based on clover results

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -331,11 +340,13 @@
                     </instructions>
                 </configuration>
             </plugin>
+            <!-- Temporarily host coveralls SNAPSHOT with clover support locally  -->
             <plugin>
                 <groupId>org.eluder.coveralls</groupId>
                 <artifactId>coveralls-maven-plugin</artifactId>
-                <version>4.3.0</version>
+                <version>4.4.0-SNAPSHOT</version>
```
Comment (lhazlewood) [inline pom.xml]: Given this is a snapshot version, will this prevent cutting a release (since snapshots are not allowed for releases?)

Comment (dogeared) [reply pom.xml]: Yep. It shouldn't be merged until (a) we either host the snapshot version somewhere (in the event the coveralls maven plugin folks drag their feet or (b) they release the 4.4.0 version of the plugin.

**Comment by dogeared** [comment]
@lhazlewood Please review again. I've hosted the `4.4.0` version of the `coveralls-maven-plugin` on bintray and referenced it from the pom as a temporary workaround until such time as it's officially released.

Once released, it will be a simple pom update to reference the official release version.

**Comment by lhazlewood** [comment]
LGTM!  Feel free to merge!

**Pull request merged**